assets/js/modules/adsense/datastore/service.js

```diff
@@ -0,0 +1,170 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Wordpress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { STORE_NAME as CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import { parseDomain } from '../util/url';
+
+const { createRegistrySelector } = Data;
+
+export const selectors = {
+	/**
+	 * Gets a URL to the service.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state        Data store's state.
+	 * @param {Object} [args]       Object containing optional path and query args
+	 * @param {string} [args.path]  A path to append to the base url.
+	 * @param {Object} [args.query] Object of query params to be added to the URL.
+	 * @return {string} The URL to the service
+	 */
+	getServiceURL: createRegistrySelector( ( select ) => ( state, { path, query } = {} ) => {
+		const userEmail = select( CORE_USER ).getEmail();
+		if ( userEmail === undefined ) {
+			return undefined;
+		}
+
+		const baseURI = 'https://www.google.com/adsense/new';
+		const queryParams = query ? { ...query, authuser: userEmail } : { authuser: userEmail };
+		if ( path ) {
+			const sanitizedPath = `/${ path.replace( /^\//, '' ) }`;
+			return addQueryArgs( `${ baseURI }${ sanitizedPath }`, queryParams );
+		}
+		return addQueryArgs( baseURI, queryParams );
+	} ),
+
+	/**
+	 * Returns the URL for creating a new AdSense account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense URL to create a new account.
+	 */
+	getCreateAccountURL: createRegistrySelector( ( select ) => () => {
+		const query = {
+			source: 'site-kit',
+			utm_source: 'site-kit',
+			utm_medium: 'wordpress_signup',
+		};
+		const siteURL = select( CORE_SITE ).getReferenceSiteURL();
+		if ( undefined !== siteURL ) {
+			query.url = siteURL;
+		}
+		return addQueryArgs( 'https://www.google.com/adsense/signup/new', query );
+	} ),
+
+	/**
+	 * Returns the URL to an AdSense account's overview page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense account overview URL.
+	 */
+	getAccountURL: createRegistrySelector( ( select ) => () => {
+		const accountID = select( STORE_NAME ).getAccountID();
+
+		if ( accountID === undefined ) {
+			return undefined;
+		}
+		return select( STORE_NAME ).getServiceURL( { path: `${ accountID }/home`, query: { source: 'site-kit' } } );
+	} ),
+
+	/**
+	 * Returns the URL to an AdSense account's site overview page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense account site overview URL.
+	 */
+	getAccountSiteURL: createRegistrySelector( ( select ) => () => {
```

Comment:
Based on my other comment, let's rename this selector to `getAccountManageSiteURL` and update its usages accordingly.